main/src/cgeo/geocaching/log/AbstractLoggingActivity.java

```diff
@@ -34,7 +34,11 @@ public boolean onCreateOptionsMenu(final Menu menu) {
 
         final SubMenu menuLog = menu.findItem(R.id.menu_templates).getSubMenu();
         for (final LogTemplate template : LogTemplateProvider.getTemplatesWithSignature()) {
-            menuLog.add(0, template.getItemId(), 0, template.getResourceId());
+            if (template.getResourceId() == 0) {
```

Comment:
Maybe make decision based on template name being nonnull (instead of resourceId). (I don't fully understand why the resourceId is needed as an attribute of the template anyway)

Reply:
This code is handling not just the new template but the whole "insert" menu. "template" can be an "old" template like the signature or elements  like Date, Owner, ... - or the new templates. The old ones have a static name being defined as resource, only the 5 new templates have a dynamic name that's provided as string.
The check could either be  (template.getResourceId() == 0) or (!template.getName().isEmpty()), same result.